src/veh_interact.cpp

```diff
@@ -1419,13 +1448,55 @@ void veh_interact::calc_overview()
         right_print( w, y, 1, c_light_gray, _( "Who" ) );
     };
 
+}
+
+std::vector<veh_interact::part_option> veh_interact::get_vehicle_overview( const vehicle *car,
+        const bool is_carried )
+{
+    std::vector<veh_interact::part_option> tmp_overview_opts;
+    const hotkey_queue &hotkeys = hotkey_queue::alphabets();
+
+    const auto next_hotkey = [&]( const vehicle_part & pt, input_event & evt ) {
+        if( overview_action && overview_enable && overview_enable( pt ) && !is_carried ) {
+            const input_event prev = evt;
+            evt = main_context.next_unassigned_hotkey( hotkeys, evt );
+            return prev;
+        } else {
+            return input_event();
+        }
+    };
+
     input_event hotkey = main_context.first_unassigned_hotkey( hotkeys );
+    bool add_seperator = is_carried;
 
-    for( const vpart_reference &vpr : veh->get_all_parts() ) {
-        if( !vpr.part().is_available() ) {
+    for( const vpart_reference &vpr : car->get_all_parts() ) {
+        if( vpr.part().is_broken() ||
+            is_carried != vpr.part().has_flag( vehicle_part::carried_flag ) ) {
             continue;
         }
 
+        if( add_seperator && !car->name.empty() && ( vpr.part().is_engine() || vpr.part().is_tank() ||
+                vpr.part().is_battery() || vpr.part().is_reactor() || vpr.part().is_turret() ||
+                vpr.part().is_seat() ) ) {
+            auto details = []( const vehicle_part & pt, const catacurses::window & w, int y ) {
+                std::string text = " <color_green>" + pt.carried_name() + "</color> ";
+
+                int max_w = ( ( TERMX - 2 ) / 3 ) - 1;
+                std::string sperator_line = std::string( std::max( static_cast<int>( max_w - 1 -
+                                            pt.carried_name().size() -
+                                            2 ), 0 ),
+                                            '=' );
+                sperator_line.insert( ( sperator_line.length() / 2 ), text );
+                right_print(
+                    w, y, 1, c_light_gray,
+                    _( sperator_line ) );
```

Comment:
Again, is this styling intentional? This looks like it would all fit on one line.